Source/JavaScriptCore/runtime/ISO8601.cpp

```diff
@@ -1680,20 +1674,84 @@ std::optional<ExactTime> ExactTime::add(Duration duration) const
+
+// https://tc39.es/proposal-temporal/#sec-validatetemporalroundingincrement
+static void validateTemporalRoundingIncrement(JSGlobalObject* globalObject, unsigned increment,
+    Int128 dividend, bool inclusive)
 {
-    Int128 incrementNs { increment };
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    Int128 maximum = 0;
+    if (inclusive)
+        maximum = dividend;
+    else {
+        ASSERT(dividend > 1);
+        maximum = dividend - 1;
+    }
+    if (increment > maximum)
+        throwRangeError(globalObject, scope, "Rounding increment exceeds maximum value"_s);
+    else if (dividend % increment)
+        throwRangeError(globalObject, scope, "Rounding increment does not divide evenly into maximum value"_s);
+}
+
+// https://tc39.es/proposal-temporal/#sec-temporal.instant.prototype.round
+// (Steps 10-17 only)
+ExactTime ExactTime::round(JSGlobalObject* globalObject, unsigned increment,
+    TemporalUnit unit, RoundingMode roundingMode) const
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    Int128 maximum = 0;
     switch (unit) {
-    case TemporalUnit::Hour: incrementNs *= ExactTime::nsPerHour; break;
-    case TemporalUnit::Minute: incrementNs *= ExactTime::nsPerMinute; break;
-    case TemporalUnit::Second: incrementNs *= ExactTime::nsPerSecond; break;
-    case TemporalUnit::Millisecond: incrementNs *= ExactTime::nsPerMillisecond; break;
-    case TemporalUnit::Microsecond: incrementNs *= ExactTime::nsPerMicrosecond; break;
-    case TemporalUnit::Nanosecond: break;
+    case TemporalUnit::Hour: maximum = (Int128) WTF::hoursPerDay; break;
+    case TemporalUnit::Minute: maximum = (Int128) (minutesPerHour * WTF::hoursPerDay); break;
+    case TemporalUnit::Second: maximum = (Int128) (secondsPerMinute * minutesPerHour * WTF::hoursPerDay); break;
+    case TemporalUnit::Millisecond: maximum = (Int128) msPerDay; break;
+    case TemporalUnit::Microsecond: maximum = (Int128) msPerDay * 1000; break;
+    case TemporalUnit::Nanosecond: maximum = nsPerDay; break;
     default:
         ASSERT_NOT_REACHED();
     }
-    return roundNumberToIncrement(quantity, incrementNs, roundingMode);
+    validateTemporalRoundingIncrement(globalObject, increment, maximum, true);
+    RETURN_IF_EXCEPTION(scope, { });
+    auto roundedNs = roundTemporalInstant(m_epochNanoseconds, increment, unit, roundingMode);
+    return ExactTime { roundedNs };
+}
+
+// https://tc39.es/proposal-temporal/#sec-temporal-roundtimedurationtoincrement
+static Int128 roundTimeDurationToIncrement(JSGlobalObject* globalObject, Int128 d, Int128 increment,
+    RoundingMode roundingMode)
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    Int128 rounded = roundNumberToIncrementInt128(d, increment, roundingMode);
+    if (absInt128(rounded) > InternalDuration::maxTimeDuration) {
+        throwRangeError(globalObject, scope, "Rounded instant exceeds maximum time duration"_s);
```

Comment:
```suggestion
        throwRangeError(globalObject, scope, "Rounded time duration exceeds maximum"_s);
```
(Or a more descriptive error message based on where this is called from, but that could be difficult as it's called from several places, not all of which correspond to a time duration that the user supplied)

Reply:
Fixed the error message in [c719f90](https://github.com/WebKit/WebKit/pull/45603/commits/c719f90b74999ab3f185aa9f6f54ad72d3749c37) . I agree that a more descriptive error message could be difficult.